### PR TITLE
gptp SYSTIM issue change

### PR DESCRIPTION
**Pull request by psow**
This change is all about making SYSTIM alterations only once in case of i210 and i211.

### CODE REVIEW

**Review by jfornal — APPROVED** [review]

**Comment by andrew-elder** [comment]
@christopher-s-hall - any comments?

File: kmod/igb_main.c [hunk 1]
```diff
@@ -134,6 +134,9 @@ static void igb_dma_err_timer(unsigned long data);
 static void igb_watchdog_task(struct work_struct *);
 static void igb_dma_err_task(struct work_struct *);
 /* AVB specific */
+#ifdef HAVE_PTP_1588_CLOCK
```
Comment (andrew-elder) [inline kmod/igb_main.c:137]: Why is this conditional? Don't all devices covered by this driver support PTP_1588_CLOCK?

**Comment by christopher-s-hall** [comment]
What does this patch accomplish? A more descriptive commit message would be helpful.

**Comment by andrew-elder** [comment]
@psow - any further comments?

**Comment by psow** [comment]
Sorry for the delay.

When the PTP daemon is running it takes control over SYSTIM (using kernel's clock_adjtime with clockid from /dev/ptpX). At this point any change made to SYSTIM by the driver can confuse the daemon and in the result break some use cases. Previously SYSTIM was changed every time the igb_reset was called (in the igb_ptp_reset). The solution is to initialize SYSTIM once in the driver and don't touch it (change it) again.

File: kmod/igb_main.c (hunk 1 above)
Comment (psow) [reply kmod/igb_main.c:137]: Old versions of the kernel don't support it. HAVE_PTP_1588_CLOCK is defined in kcompat.h mostly based on kernel version and configuration.

**Comment by psow** [comment]
@christopher-s-hall @andrew-elder - do you have any other questions/doubts?

**Comment by andrew-elder** [comment]
@psow - can you please redo the commit message text to contain the background you wrote above. @christopher-s-hall - any further concerns? I'm ok with this.

**Comment by psow** [comment]
> @psow - can you please redo the commit message text to contain the background you wrote above.

Done.

**Comment by andrew-elder** [comment]
Thanks!

**Comment by psow** [comment]
@christopher-s-hall ping :)

**Comment by christopher-s-hall** [comment]
This is OK. I think we still might have a problem maintaining the clock over a hardware reset, i.e. SYSTIM becomes 0, but that's probably something we can fix-up in the daemon.

**Review by christopher-s-hall — APPROVED** [review]

**Pull request merged**